### DISCUSSION TITLE
Creating a map with hills

### DISCUSSION THREAD

**Discussion by NnaYelsel** [carla-simulator/carla#7284] [Q&A]
Hello everyone,

First, note that I'm using **Carla v0.9.15 on Windows**, build from source.

**I'm looking to create a basic map with slopes on the roads, and some buildings and trees around.** 

I read the documentation but my feeling is that the documentation about map creation / customization is spread around the carla website. I'm confused. I noticed these pages:
_- https://carla.readthedocs.io/en/stable/map_customization/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_generate_map/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_add_map_source/
- https://carla.readthedocs.io/en/latest/adv_digital_twin/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_custom_layers/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_custom_add_tl/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_custom_buildings/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_custom_road_painter/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_custom_weather_landscape/
- https://carla.readthedocs.io/en/0.9.15/tuto_M_custom_weather_landscape/_

I'm trying to put everything together and I think that there is 3 ways to make it:

**1 - using the Digital Twin Tool** 
**2 - using OSM/XODR/RoadRunner/Carla import**
**3 - duplicate an existing town and modify it**

Let's talk about each solution:
**1 - using the Digital Twin Tool** 
-> UW_OnRoadMainWidget makes UE crash when I'm trying to generate something

**2 - using OSM/XODR/RoadRunner/Carla import**
-> I achieve to convert OSM to XODR files and show it with config -x command but it's seems that all the slopes disappear. 
Then I don't have RoadRunner license anyway (I could buy one but want to be sure that after it, I could reach my goal).

**3 - duplicate an existing town and modify it**
Tried to duplicate Town1 (like in  https://carla.readthedocs.io/en/stable/map_customization/) but for me "CarlaMapGenerator" is just a folder and I can't find any seed related to it. By the way, I tried to modify slightly the road but can't access to any map of road slope. I can just modify the elevation of a whole piece of road.

Then I'm stuck, so if someone can give me some feedback and advice. Which one of these three options should I dig in ? 
Are some pages in the documentation much adapted to my problem than others ? 

Does someone have a sample file given from roadrunner to give to me (i.e a .fbx) ? just for me to dig the solution 2. 

Is a future release planned that could help me ? 

Thank for reading my question.

Best regards.

**Comment by anshgwash** [comment]
OSM doesn't support elevation which is why all the slopes disappear when you try to convert from OSM to XODR. You could try and add elevation information back into the XODR using another converter or custom tool. https://github.com/JHMeusener/osm2xodr has support for elevation if you supply a topo map.
